src/plugins/procmon/procmon.cpp

```diff
@@ -145,6 +145,41 @@ struct process_visitor_ctx
 
 } // namespace
 
+static char* read_cmd_line(vmi_instance_t vmi, drakvuf_trap_info_t* info, addr_t addr)
+{
+    char *cmd = NULL;
+    access_context_t ctx2 =
+            {
+                    .translate_mechanism = VMI_TM_PROCESS_DTB,
+                    .dtb = info->regs->cr3,
+                    .addr = addr,
+            };
+    addr_t cmdline_addr = ctx2.addr;
+    uint16_t cmd_len = 0;
+    if(VMI_SUCCESS == vmi_read_16(vmi,&ctx2,&cmd_len))
+    {
+        ctx2.addr = cmdline_addr+8; // _UNICODE_STRING->Buffer
+        addr_t buffer_adr = 0;
+        if(VMI_SUCCESS == vmi_read_addr(vmi,&ctx2,&buffer_adr))
+        {
+            ctx2.addr = buffer_adr;
+            char *buf_ret;
+            buf_ret = (char*)g_try_malloc0(cmd_len+1);
+            if (!buf_ret) return NULL;
+            if(VMI_SUCCESS == vmi_read(vmi,&ctx2,cmd_len,buf_ret,NULL))
+            {
+                cmd = (char*)g_try_malloc0(cmd_len+1);
+                if (!cmd) return NULL;
```

Comment:
I think you are leaking `buf_ret` if you just return NULL here.

Reply:
You marked it resolved but I don't see it being freed?

Reply:
Is it not appropriate to freed buf_ret here?
[#L179](https://github.com/tklengyel/drakvuf/blob/caa0487d633f101745315a06a4d9a1f39c8424bd/src/plugins/procmon/procmon.cpp#L179) 
@tklengyel

Reply:
If `buf_ret` allocation succeeds but `cmd` not, then `cmd` is `NULL` and you will exit the function here:

https://github.com/tklengyel/drakvuf/blob/caa0487d633f101745315a06a4d9a1f39c8424bd/src/plugins/procmon/procmon.cpp#L172

Thus, line 179 will not be reached and you leak the memory.

Reply:
Is it okay?
[d0df43e](https://github.com/tklengyel/drakvuf/pull/833/commits/d0df43ea7279ac9d28bdd9edf1c27f8aaa1e241b)

Reply:
```cpp
                if (!cmd){
                    g_free(buf_ret);
                    g_free(cmd);
                    return NULL;
                }
```

You can remove `g_free(cmd)` - if you are inside that `if` then `cmd` is guaranteed to be `NULL`  - unallocated.